### ISSUE TITLE
Loss doesn't require grad. Did you define any loss in the task?

### ISSUE THREAD

**Issue by haifangong** [DeepGraphLearning/torchdrug#55]
I have defined the mse loss but it's still useless.

Here comes the codes:
```
train_set, valid_set, test_set, dataset = load_dataset('Caco-2-Permeability.csv')
model = models.GIN(input_dim=dataset.node_feature_dim,
                   hidden_dims=[256, 256, 256, 256],
                   short_cut=True, batch_norm=True, concat_hidden=True)
task = tasks.PropertyPrediction(model, task=(),
                                criterion="mse", metric="rmse")
optimizer = torch.optim.Adam(task.parameters(), lr=1e-3)
solver = core.Engine(task, train_set, valid_set, test_set, optimizer,
                     gpus=[0], batch_size=1024)
solver.train(num_epoch=100)
solver.evaluate("valid")
solver.save("clintox_gin_infograph.pth")
```

**Comment by haifangong** [comment]
I have defined the mse loss but it's still useless.

**Comment by haifangong** [comment]
Here comes the codes:
    train_set, valid_set, test_set, dataset = load_dataset('Caco-2-Permeability.csv')
    model = models.GIN(input_dim=dataset.node_feature_dim,
                       hidden_dims=[256, 256, 256, 256],
                       short_cut=True, batch_norm=True, concat_hidden=True)
    task = tasks.PropertyPrediction(model, task=(),
                                    criterion="mse", metric="rmse")
    optimizer = torch.optim.Adam(task.parameters(), lr=1e-3)
    solver = core.Engine(task, train_set, valid_set, test_set, optimizer,
                         gpus=[0], batch_size=1024)
    solver.train(num_epoch=100)
    solver.evaluate("valid")
    solver.save("clintox_gin_infograph.pth")

**Comment by KiddoZhu** [comment]
Hi! It looks like you don't define any task for the property prediction. So there is no loss for the training, and the error is correct. You need to add the name of the field of the labels as a task when creating `tasks.PropertyPrediction`.

**Comment by haifangong** [comment]
Thank you for your quick response.
